vision/blob/blobfind.py

```diff
@@ -0,0 +1,105 @@
+"""
+Detects blobs in images using OpenCV's SimpleBlobDetector.
+"""
+import os
+import sys
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+import cv2
+from vision.blob.blob import Rectangle
+import json
+
+def import_params():
+    params = cv2.SimpleBlobDetector_Params()
+
+    with open(os.path.join('vision', 'blob', 'config.json'), 'r') as config_file:
+        config = json.load(config_file)
+        config_file.close()
+
+        for category in config:
+            category_enabled = config[category]['enable']
+            if category_enabled:
+                if hasattr(params, category):
+                    setattr(params, category, config[category]['enable'])
+                for attr in config[category]:
+                    if hasattr(params, attr):
+                        setattr(params, attr, config[category][attr])
+
+        print(params.minThreshold)
+    return params
+
+
+class BlobFinder:
+    """
+    Constructs a BlobFinder object with an image, logging, and params
+
+    Parameters
+    ----------
+    image: np array
+        image to detect blobs in, as a np array
+    params: SimpleBlobDetector_Params
+        blob detector params object
+    """
+    def __init__(self, image, params=None):
+        self.image = image
+        self.keypoints = []
+        if params is None:
+            self.params = import_params()
+        else:
+            self.params = params
+
+    def update_image(self, image):
+        self.image = image
+
+    def update_params(self, params):
+        self.params = params
```

Comment:
Lets use properties for these ie pythons take on getters and setters.
```
@property
def image(self):
    """
    This function is called every time self.image is run.
    """
    return self._image  # You shouldn't need to rename any instances of image to _image, the getter and setters will work.

@image.setter
def image(self, value):
    """
    Defines behavior of self.image = value.
    """
    if not instance(value, np.ndarray):
        raise ValueError("Need an image")
    self._image = image
```
Same for params